pczt/Cargo.toml

```diff
@@ -11,3 +11,67 @@ license.workspace = true
 categories.workspace = true
 
 [dependencies]
+zcash_note_encryption = { workspace = true, optional = true }
+zcash_primitives = { workspace = true, optional = true }
+zcash_protocol = { workspace = true, optional = true }
+
+blake2b_simd = { workspace = true, optional = true }
+rand_core = { workspace = true, optional = true }
+
+# Encoding
+postcard = { version = "1", features = ["alloc"] }
+serde.workspace = true
+serde_with = "3"
+
+# Payment protocols
+# - Transparent
+secp256k1 = { workspace = true, optional = true  }
+
+# - Sapling
+bls12_381 = { workspace = true, optional = true }
+ff = { workspace = true, optional = true }
+jubjub = { workspace = true, optional = true }
+redjubjub = { workspace = true, optional = true }
+sapling = { workspace = true, optional = true }
+
+# - Orchard
+nonempty = { workspace = true, optional = true }
+orchard = { workspace = true, optional = true }
+pasta_curves = { workspace = true, optional = true }
+
+[dev-dependencies]
+incrementalmerkletree.workspace = true
+secp256k1 = { workspace = true, features = ["rand"] }
+shardtree.workspace = true
+zcash_primitives = { workspace = true, features = ["test-dependencies",  "transparent-inputs"] }
+zcash_proofs = { workspace = true, features = ["bundled-prover"] }
+zip32.workspace = true
+
+[features]
+orchard = [
+    "dep:ff",
+    "dep:nonempty",
+    "dep:orchard",
+    "dep:pasta_curves",
+    "dep:zcash_protocol",
+]
+sapling = [
+    "dep:bls12_381",
+    "dep:ff",
+    "dep:jubjub",
+    "dep:redjubjub",
+    "dep:sapling",
+    "dep:zcash_note_encryption",
+    "dep:zcash_protocol",
+]
+transparent = ["dep:secp256k1", "dep:zcash_primitives", "dep:zcash_protocol"]
+zcp-builder = ["dep:zcash_primitives", "dep:zcash_protocol"]
+io-finalizer = ["orchard", "sapling"]
+prover = ["dep:rand_core", "sapling?/temporary-zcashd"]
+signer = ["dep:blake2b_simd", "dep:rand_core", "orchard", "sapling", "transparent"]
+spend-finalizer = ["transparent"]
+tx-extractor = ["dep:rand_core", "orchard", "sapling", "transparent"]
```

Comment:
These feature flags are not sufficiently granular, but I don't know what the right way is to do it - for example, you can't use `zcash_client_backend` with the `orchard` flag disabled but also use `pczt/signer` because the latter pulls in the `orchard` dependencies.

Reply:
For now I think that I'm going to make `zcash_client_backend/pczt` imply `orchard` and `transparent-inputs` and then we can worry about other combinations later.

Reply:
Yep. The granularity here is limited due both to `zcash_primitives` not having flags yet, and due to the transparent protocol living in that crate. It causes a bit of a mess when you're trying to either use transparent or any role that requires multiple protocols.